compiler/src/scala/quoted/runtime/impl/QuoteMatcher.scala

```diff
@@ -452,20 +433,52 @@ object QuoteMatcher {
       accumulator.apply(Set.empty, term)
   }
 
+  enum MatchResult:
+    case ClosedTree(tree: Tree)
+    case OpenTree(tree: Tree, patternTpe: Type, args: List[Tree], env: Env)
+
+    def toExpr(mapTypeHoles: TypeMap)(using Context): Expr[Any] = this match
+      case MatchResult.ClosedTree(tree) =>
+        new ExprImpl(tree, SpliceScope.getCurrent)
+      case MatchResult.OpenTree(tree, patternTpe, args, env) =>
+        def hoasClosure = {
```

Comment:
Why a def if we then immediately call it?

Reply:
Accident of refactoring.